pywr/parameters/_parameters.pyx

```diff
@@ -594,6 +608,28 @@ cdef class IndexParameter(Parameter):
         """Returns the current index"""
         # return index as an integer
         return 0
+
+    cpdef setup(self):
+        super(IndexParameter, self).setup()
+        cdef int num_comb
+        if self._model.scenarios.combinations:
+            num_comb = len(self._model.scenarios.combinations)
+        else:
+            num_comb = 1
+        self.__indices = np.empty([num_comb], np.int32)
+
+    cpdef calc_values(self, Timestep timestep):
+        cdef ScenarioIndex scenario_index
+        for scenario_index in self._model.scenarios.combinations:
+            self.__values[<int>(scenario_index.global_id)] = self.value(timestep, scenario_index)
+            self.__indices[<int>(scenario_index.global_id)] = self.index(timestep, scenario_index)
```

Comment:
I wonder if the index should be calculated before the value? The value tends to depend on the index.